typespec-extension/src/emitter.ts

```diff
@@ -49,6 +49,7 @@ export interface DevOptions {
   "support-versioning"?: boolean;
   "debug"?: boolean;
   "loglevel"?: "off" | "debug" | "info" | "warn" | "error";
+  "java-temp-dir"?: string; // working directory for java codegen, e.g. transformed code-model file
```

Comment:
Does user or dev need to use this config?

Reply:
Current codegen puts preprocessed code-model under `java.io.tmpdir`, which is very tricky to find for linux-based OS.
https://github.com/Azure/autorest.java/blob/ecf3d350fd1cd59459965d435d2ce30945cf5ab4/fluentnamer/src/main/java/com/azure/autorest/fluentnamer/FluentNamer.java#L63

 It also provides a way to override the default tmp dir:
https://github.com/Azure/autorest.java/blob/ecf3d350fd1cd59459965d435d2ce30945cf5ab4/extension-base/src/main/java/com/azure/autorest/extension/base/util/FileUtils.java#L33

This `java-temp-dir` is equivalent to the `codegen.java.temp.directory` in codegen.